src/material/autocomplete/autocomplete-trigger.ts

```diff
@@ -511,7 +521,7 @@ export abstract class _MatAutocompleteTriggerBase
    * stream every time the option list changes.
    */
   private _subscribeToClosingActions(): Subscription {
-    const firstStable = this._zone.onStable.pipe(take(1));
+    const firstStable = this._zone.onStable.pipe(take(1), this._enterZone());
```

Comment:
Rather than creating our own operator, wouldn't it be easier to wrap the relevant logic in `NgZone.run`? It seems like it's only the logic inside the `switchMap` that needs it.

Reply:
Hey. I didn't place it inside the `switchMap` and created the `_enterZone` method (I've taken it from other parts of the code btw), because the `onStable` emits outside of the Angular zone, but `optionChanges` emits within. So that would cause redundant change detection if we call `ngZone.run` inside the `switchMap`.

Reply:
My understanding is that if something is already in the zone, calling `run` again won't trigger another change detection.

Reply:
You're right, I've updated it, had to re-check it.